### ISSUE TITLE
OpenAudible no longer finds my downloaded books

### ISSUE THREAD

**Issue by IamMaris** [openaudible/openaudible#1046]
I updated  to 3.6.2 today and when I opened the app on my PC, it no longer found my books. I checked the preferences to verify the file structure and it is correct. The folders do have the aax books, the mp3 file I converted, and art work, a relief because at first I thought they were deleted somehow.

I have re-entered the file structure, closed and re-opened the application a few times, but it still opens with a blank screen and no indication of the hundreds of books I have or the 80 or so mp3 converted books I have. How do I get it to read my library again?

My PC is a Samsung Book 360 funning Windows 11 Home, Version 21H2, and I have been using OpenAudible on it successfully  until today.

**Comment by openaudible** [comment]
Under the Help menu is a Diagnostics. Maybe your library paths out of whack. Make sure you have a “books” file path and it exists and has your books.

**Comment by IamMaris** [comment]
Thank you for the response. I ended up renaming my folders for books and pdfs to preserve them, and then reinstalling the app again and then seeing what it used for file structure. It seemed to duplicate what I had been using. For example, it saves my downloads into a folder named aax, but instead of where the folder is supposed to be, it creates another folder named aax inside the original aax, and places the downloaded book files there.  I downloaded one book to see what happened and that is how I discovered it. 

When I then converted that downloaded book to mp3 to trace its location, it placed it into a folder called books, but again, inside the books folder is another folder called books where the mp3 ended up.  Once I knew where it was putting things, I imported my previously downloaded books into those duplicate named folders. I was missing 242 books, and had to track down what was missing. They were there, but  OA couldn't recognize that. To figure that out, I downloaded one of the missing books and discovered the name was slightly different from my existing book. I then tried another refresh which after a while opened a window that said it discovered 242 books that were not included, should it add them. Of course, I said yes, and after a while watching it "import" them, it completed and said it was saving and updating cache. However, I noticed when that was done, I was still missing the same 242 books which are there in the same folder, just not discovered by OA. Additional refreshes did not make a difference, just repeating the same message, claiming to import and update and save, but with the same missing results. I also have 28 books with missing links, so I will probably never get those back, as from reading the help files it appears they may be too old and no longer available on Audible. If you know how to address that probem, it would be nice to get them recognized again.

Do I really have to re-download over 200 books, or is there something I am not doing that will force the app to see the files that are actually there but not recognized? Oh, and for the mp3 files, I just copied them over to where it put the test mp3 and it found all of them without any problem. It seems to be the actual book files that aren't behaving. (None are downloaded in sections)

I have had this app for a long time, updated it a few times, and have never run into this before after an update. 

Maris

**Comment by IamMaris** [comment]
OK, I'm pretty much back to normal. I ended up moving my original downloads to another area, Reinstalled OpenAudible, and imported all my books from my original folder. That worked for almost all of the 1273 books except for 103, which I just downloaded again.  

I have no idea what happened during the update that went wrong, but it's sorted now, even for the few that were no longer available to download and had "missing links" that this latest import found and imported.  

Closing the issue.
